packages/demo/src/app/connected/(tools)/TimeLockedTransfer/page.tsx

```diff
@@ -0,0 +1,224 @@
+  outPoint: {
+    txHash: "0x54090200597d544f0c8e8c02a0dc12715260ccc88dfe097ae76198ba71f71bff",
+    index: 0,
+  },
+  depType: "code",
+}
+const feeRate = 3000;
+
+export default function TimeLockedTransfer() {
+  const { signer, createSender } = useApp();
+  const { log, error } = createSender("Transfer");
+
+  const { explorerTransaction } = useGetExplorerLink();
+
+  const [transferTo, setTransferTo] = useState<string>("");
+  const [amount, setAmount] = useState<string>("");
+  const [data, setData] = useState<string>("");
+  const [lockedForBlocks, setLockedForBlocks] = useState<string>("");
+  const [liveTimeLockCell, setLiveTimeLockCell] = useState<ccc.Cell | null>(null);
+  const [showClaimPage, setShowClaimPage] = useState<boolean>(false);
+  const [claimData, setClaimData] = useState<string>("");
+
+  const handleTimeLockedTransfer = async () => {
+    if (!signer) {
+      return;
+    }
+
+    // Verify destination addresses
+    const toAddress = await ccc.Address.fromString(transferTo, signer.client);
+
+    const currentBlockNumber = await signer.client.getTip();
+    const lockedUntil = new Since("absolute", "blockNumber", currentBlockNumber + numFrom(lockedForBlocks));
+    const lockedUntilToNum = lockedUntil.toNum();
+    const timeLockScript = ccc.Script.from({
+      ...testnetTimeLockScript,
+      args: buildTimeLockArgs(toAddress.script.hash(), lockedUntilToNum.toString()),
+    })
+
+    const timeLockCellOutput = { lock: timeLockScript };
+
+    const tx = ccc.Transaction.from({
+      outputs: [timeLockCellOutput],
+      outputsData: [bytesFromAnyString(data)],
+    });
+
+    const minimumCapacity = tx.getOutputsCapacity();
+    if (minimumCapacity > ccc.fixedPointFrom(amount)) {
+      error(`Insufficient capacity to store data`);
+      return;
+    }
+    tx.outputs[0].capacity = ccc.fixedPointFrom(amount);
+
+    // Complete missing parts for transaction
+    await tx.completeInputsByCapacity(signer);
+    await tx.completeFeeBy(signer, feeRate);
+
+    signer.sendTransaction(tx).then((hash: string) => {
+      log(
+        "Transaction sent:",
+        explorerTransaction(hash),
+      );
+    }).catch((err: Error) => {
+      error(err.message);
+    });
+  };
+
+  useEffect(() => {
+    checkTimeLockedCells();
+  }, []);
+
+  const checkTimeLockedCells = async () => {
+    if (!signer) {
+      return;
+    }
+    // cell ordering is not guaranteed
+    const timeLockedCells: ccc.Cell[] = [];
+    for await (const cell of signer.client.findCells(
+      {
+        script: ccc.Script.from({
+          ...testnetTimeLockScript,
+          args: bytesFromAnyString((await signer.getRecommendedAddressObj()).script.hash()),
+        }),
+        scriptType: "lock",
+        scriptSearchMode: "prefix",
+      },
+      "desc",
+    )) {
+      // Process each cell
+      timeLockedCells.push(cell);
+    }
+    // pick the first one
+    setLiveTimeLockCell(timeLockedCells[0]);
```

Comment:
If you only need the first cell, setting the cell and breaking the loop will be better.
Also, check [Nervos DAO](https://github.com/ckb-ecofund/ccc/blob/master/packages/demo/src/app/connected/(tools)/NervosDao/page.tsx) if you want to list all cells.

Reply:
Yeah, the first cell is just enough for the current workflow.